### PR TITLE
Tooltip path

### PR DESCRIPTION
**Pull request by jread-usgs**
This is an add-on to #282 , but makes the tooltip box as a single <path> instead of a <rect> (for the box) and a <path> for the triangle tip. 

This gets rid of the little bit of separation or overlap between the triangle and the box that appears in other vizzies: 
image

### CODE REVIEW

**Comment by dblodgett-usgs** [comment]
👍 I pulled down your branch on top of the latest here and it works great!

screen shot 2017-10-15 at 10 26 59 am

File: inst/js/tooltip.js [hunk 1]
```diff
@@ -1,44 +1,69 @@
...
+    var textBuffer;
     var tooltipX;
-    var tooltip_bg = tipG.append('rect').attr("id", "tooltip-box").attr("height", 24).attr("class", "tooltip-box");
-    var tool_pt = tipG.append('path').attr("id", "tooltip-point").attr("d", "M-6,-11 l6,10 l6,-11").attr("class","tooltip-box");
-    var tooltip = tipG.append('text').attr("id", "tooltip-text").attr("dy","-1.1em").attr('text-anchor',"middle").attr("class","tooltip-text-label svg-text").text(text);
+    var tooltip_bg;
+    var tooltip;
+    tooltip_bg= tipG.append('path').
+      attr("id", "tooltip-box").
+      attr("class", "tooltip-box");
```
Comment (aappling-usgs) [inline inst/js/tooltip.js]: good idea to get this on multiple lines.

the convention i've been seeing is to put the dot at the beginning of each new line (e.g. https://d3js.org/#introduction) - do you have an opinion on where the dots should go, and why?

also, it looks to me like there's nothing stopping you from continuing to declare and define each variable in a single command, which seems a little cleaner to me. Like this:
```
var tooltip_bg = tipG.append('path')
    .attr("id", "tooltip-box")
    .attr("class", "tooltip-box");
```
what do you think about doing it that way instead? are there reasons not to?

File: inst/js/tooltip.js [hunk 2]
```diff
@@ -1,44 +1,69 @@
...
+      text(text);
     
-    textLength = Math.round(tooltip.node().getComputedTextLength());
+    textBox = tooltip.node().getBBox();
+    textLength = Math.round(textBox.width);
+    textHeight = Math.round(textBox.height);
     halfLength = textLength / 2;
+    textBuffer = 6;
+    var tipYoffset = -2; // so that the tip is slightly above the mouse location
+    var tipTriangle = {x:6, y:10};
```
Comment (aappling-usgs) [inline inst/js/tooltip.js]: good idea to extract & name all these constants

File: inst/js/tooltip.js (hunk 1 above)
Comment (jread-usgs) [reply inst/js/tooltip.js]: I was following Mary's style for the `var varname;` style ones. But the PR is a mix of both. Not sure why to do it in two lines?

Comment (jread-usgs) [reply inst/js/tooltip.js]: I also prefer the .attr() line starts. I will clean that up.

**Pull request merged**